test/Microsoft.DotNet.TestFramework/Commands/TestCommand.cs

```diff
@@ -243,6 +243,17 @@ private void AddEnvironmentVariablesTo(ProcessStartInfo psi)
 
             //  Flow the TEST_PACKAGES environment variable to the child process
             psi.Environment["TEST_PACKAGES"] = RepoDirectoriesProvider.TestPackages;
+
+            // Set DOTNET_ROOT to ensure sub process find the same host fxr
+            string dotnetDirectoryPath = Path.GetDirectoryName(RepoDirectoriesProvider.DotnetUnderTest);
+            if (System.Environment.Is64BitProcess)
+            {
+                psi.Environment.Add("DOTNET_ROOT", dotnetDirectoryPath);
```

Comment:
@dsplaisted FYI I added DOTNET_ROOT to all test command. I realize if you run _dotnet run_, it targets stage 0